apps/projects/contracts/Projects.sol

```diff
@@ -287,6 +312,101 @@ contract Projects is AragonApp {
+        uint256 _issueNumber, 
+        string _application
+    ) external isInitialized 
+    {
+        require(bytes(repos[_repoId].issues[_issueNumber].assignmentRequests[msg.sender]).length == 0, "User already applied for this issue");
+
+        repos[_repoId].issues[_issueNumber].applicants.push(msg.sender);
+        repos[_repoId].issues[_issueNumber].assignmentRequests[msg.sender] = _application;
+
+        emit AssignmentRequested(msg.sender, _repoId, _issueNumber);
+    }
+
+    /**
+     * @notice approve a request for assignment to a single requestor
+     * @param _repoId the github repo id of the issue
+     * @param _issueNumber the github issue up for assignment
+     * @param _requestor address of user that will be assigned the issue
+     */
+    function approveAssignment(
+        bytes32 _repoId, 
+        uint256 _issueNumber, 
+        address _requestor
+    ) external isInitialized auth(TASK_MGR_ROLE)
+    {
+        require(bytes(repos[_repoId].issues[_issueNumber].assignmentRequests[_requestor]).length != 0, "User has not applied for this issue");
+        repos[_repoId].issues[_issueNumber].assignee = _requestor;
+
+        emit AssignmentApproved(_requestor, _repoId, _issueNumber);
+    }
+
+    /**
+     * @notice Submit work for issue '`_issueNumber`'.
+     * @dev add a submission to local state after it's been added to StandardBounties.sol
+     * @param _repoId the github repo id of the issue
+     * @param _issueNumber the github issue up for assignment
+     * @param _submissionAddress IPFS hash of the Pull Request
+     * @param _fulfillmentId retrieved from event after the work is submitted to the bounties contract externally 
+     */
+    function submitWork(
+        bytes32 _repoId, 
+        uint256 _issueNumber, 
+        string _submissionAddress,
+        uint256 _fulfillmentId
+    ) external isInitialized
+    {
+        require(msg.sender == repos[_repoId].issues[_issueNumber].assignee, "User not assigned to this issue");
+
+        repos[_repoId].issues[_issueNumber].workSubmittors.push(msg.sender);
+        repos[_repoId].issues[_issueNumber].workSubmissions[msg.sender] = WorkSubmission(
+            SubmissionStatus.Unreviewed,
+            _submissionAddress,
+            _fulfillmentId
+        );
+
+        emit WorkSubmitted(msg.sender, _repoId, _issueNumber);
+    }
+
+    /**
+     * @notice Review work submitted by '`_contributor`'.
+     * @dev add a submission to local state after it's been added to StandardBounties.sol
+     * @param _repoId the github repo id of the issue
+     * @param _issueNumber the github issue up for resolution
+     * @param _contributor address of the asignee who submitted work for review
+     * @param _approved decision to accept the contribution 
+     */
+    function reviewSubmission(
+        bytes32 _repoId, 
+        uint256 _issueNumber, 
+        address _contributor,
+        bool _approved
+    ) external isInitialized auth(TASK_MGR_ROLE)
```

Comment:
Do we want to split this into two roles?

Reply:
yeah that makes sense. Someone who reviews work might be have different permissions than someone who accepts